src/test/java/com/getindata/connectors/http/internal/sink/httpclient/status/ComposeHttpStatusCodeCheckerTest.java

```diff
@@ -0,0 +1,135 @@
+package com.getindata.connectors.http.internal.sink.httpclient.status;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.flink.util.StringUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.getindata.connectors.http.internal.config.HttpConnectorConfigConstants;
+
+class ComposeHttpStatusCodeCheckerTest {
+
+    private static final String STRING_CODES = "403, 100,200, 300, , 303 ,200";
+
+    private static final List<Integer> CODES =
+        Arrays.stream(STRING_CODES.split(HttpConnectorConfigConstants.ERROR_CODE_DELIM))
+            .filter(code -> !StringUtils.isNullOrWhitespaceOnly(code))
+            .map(String::trim)
+            .mapToInt(Integer::parseInt)
+            .boxed()
+            .collect(Collectors.toList());
+
+    private ComposeHttpStatusCodeChecker codeChecker;
+
+    @BeforeAll
+    public static void beforeAll() {
+        assertThat(CODES).isNotEmpty();
+    }
+
+    private static Stream<Arguments> propertiesArguments() {
+        return Stream.of(
+            Arguments.of(new Properties()),
+            Arguments.of(prepareProperties("", "")),
+            Arguments.of(prepareProperties(" ", " ")),
+            Arguments.of(prepareProperties(",,,", ",,,,"))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("propertiesArguments")
+    public void shouldPassOnDefault(Properties properties) {
+        codeChecker = new ComposeHttpStatusCodeChecker(properties);
+
+        assertAll(() -> {
+            assertThat(codeChecker.isErrorCode(100)).isFalse();
+            assertThat(codeChecker.isErrorCode(200)).isFalse();
+            assertThat(codeChecker.isErrorCode(500)).isTrue();
+            assertThat(codeChecker.isErrorCode(501)).isTrue();
+            assertThat(codeChecker.isErrorCode(400)).isTrue();
+            assertThat(codeChecker.isErrorCode(404)).isTrue();
+        });
+    }
+
+    @Test
+    public void shouldParseWhiteList() {
+
+        Properties properties = new Properties();
+        properties.setProperty(
+            HttpConnectorConfigConstants.HTTP_ERROR_CODE_WHITE_LIST,
+            STRING_CODES);
+        properties.setProperty(
+            HttpConnectorConfigConstants.HTTP_ERROR_CODES_LIST,
+            "1XX, 2XX, 3XX, 4XX, 5XX"
+        );
+
+        codeChecker = new ComposeHttpStatusCodeChecker(properties);
+
+        assertAll(() -> CODES.forEach(code -> assertThat(codeChecker.isErrorCode(code)).isFalse()));
+    }
```

Comment:
I would see a simple check that some codes different than `CODES` are considered errors.

Reply:
Added

Reply:
Thanks!